Branch-SDK-TestBed/src/main/java/io/branch/branchandroiddemo/MainActivity.java

```diff
@@ -471,6 +471,11 @@ public void onInitFinished(BranchUniversalObject branchUniversalObject, LinkProp
     @Override
     public void onNewIntent(Intent intent) {
         this.setIntent(intent);
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
         Branch.getInstance().reInitSession(this, new BranchReferralInitListener() {
```

Comment:
`reInitSession()` should not be called from `onResume()`.  It should only be called from `onNewIntent()`

Reply:
Discussed in person, we hope to remove `ForceNewBranchSession` in the future, so if that guard is not there, we should surely not call `reInitSession` from `onResume`.